contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,190 @@
+
+One-way data binding (used in React) involves uni-directional communication. If the model is changed, the changes are reflected in the UI. However, if the UI element changes the model state is not updated. Developers must provide this synchronization code. 
+
+This [article](https://medium.com/@preethi.s/angular-custom-two-way-data-binding-3e618309d6c7) on Medium provides a good introduction to two-way data binding in Angular.
+
+### When not to use Angular?
+
+Like any other framework/library, Angular has its share of disadvantages.
+
+1. **Steep Learning curve** - Angular requires you to learn many concepts, such as directives, modules, components, services, dependency injection, pipes, and most importantly TypeScript. 
+The large number of new concepts can be confusing to newcomers.
+
+2. **Opinionated Framework** - Angular is opinionated about how you structure your code. This means it provides you with a set of tools to build your application. Angular expects you to use these tools as much as you can. While libraries like React allow you to integrate any third party services in your application.
+
+You can refer to this [article](https://jsreport.io/why-is-react-more-popular-than-angular/) that points out the benefits other tools like React have over Angular.
+    
+### Angular when compared with other popular frontend frameworks
+
+There are alternate frontend frameworks and libraries that developers may prefer over Angular. <a href="https://reactjs.org/">React</a>, a Javascript library, and <a href="https://vuejs.org/">Vue</a>, a Javascript framework, are two such examples.
+
+React is a popular Javascript library, open sourced by Facebook. The ease of learning React is a key advantage.
+It also provides more flexibility to developers as it allows integration of third party libraries.
+Similarly, Vue offers us some advantages, like flexibility, simplistic structure and ease of integration.
+
+Every framework/library has its own pros and cons. The framework/library you choose depends on the requirements of the application. Below are a few resources that give a brief comparison between the popular frontend frameworks/libraries.
+
+1. [React vs Angular vs Vue](https://medium.com/@TechMagic/reactjs-vs-angular5-vs-vue-js-what-to-choose-in-2018-b91e028fa91d) - A brief comparison between the three most popular frontend tools.
+2. [Why use Vue](https://medium.com/@brainmobi/advantages-of-using-vue-js-for-your-web-applications-7e460cadfffc) - Advantages of using Vue.
+3. [Angular vs React](https://programmingwithmosh.com/react/react-vs-angular/) - A comprehensive comparison between React and Angular.
+
+## How to get started with Angular?
+
+Angular is not the easiest framework to work with. However, it is reliable and is used by many developers. Given below are a few steps that will help you ease into the Angular environment.
```

Comment:
`However, it is reliable and is used by many developers.` No need for this here. This is not the WHY section.